types/promise-hash/index.d.ts

```diff
@@ -0,0 +1,17 @@
+// Type definitions for promise-hash 1.3
+// Project: https://github.com/mtimofiiv/promise-hash
+// Definitions by: Michael Shafir <https://github.com/mshafir>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = hash;
+
+declare global {
+    interface PromiseConstructor {
+        hash<T>(promiseHash: { [P in keyof T]: PromiseLike<T[P]> | T[P] }):
```

Comment:
You can store the function type and reuse it here as well as for the declared function below.

Reply:
I'm not sure how best to do that. The most I could figure out is a shared type for the arguments with
```ts
type PromiseHash<T> = { [P in keyof T]: PromiseLike<T[P]> | T[P] };
```
I couldn't figure out how to get the whole function type to be reused.